src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -694,7 +684,7 @@ public boolean refreshNeeded() {
             try {
                 // we are either dirty due to a document added or due to a
```

Comment:
This comment is a bit stale, e.g. says "dirty" (which is gone) and e.g. deletes can also mean we are not current ... I think just remove the comment?